docs/csharp/tutorials/ranges-indexes.md

```diff
@@ -77,7 +77,11 @@ Indexes and ranges provide clear, concise syntax to access a single element or a
 Any type that provides an [indexer](../programming-guide/indexers/index.md) with an <xref:System.Index> or <xref:System.Range> parameter explicitly supports indices or ranges respectively. An indexer that takes a single <xref:System.Range> parameter may return a different sequence type, such as <xref:System.Span%601?displayProperty=nameWithType>.
 
 > [!IMPORTANT]
-> The range operator often allocates a copy to store the range of elements returned. This causes extra allocations and copies, impacting performance.
+> The performance of code using the range operator depends on the source collection operand.
+>
+> The time complexity of the range operator depends on the storage type. For example, if the storage is a `string` or an array, then the result is a copy of the specified section of the input, so the time complexity is *O(N)* (where N is the length of the range). On the other hand, if it's a <xref:System.Memory%601.Span%2A?displayProperty=nameWithType> or a <xref:System.Memory%601?displayProperty=nameWithType>, the result references the same backing store, which means there is no copy and the operation is *O(1)*.
+>
+> In addition to the time complexity, this causes extra allocations and copies, impacting performance. In performance sensitive code, consider using `Span<T>` or `Memory<T>` as the sequence type, since the range operator does not allocate for them.
```

Comment:
This now talks at different points about "source collection", "storage type" and "sequence type" and I'm not sure it will be clear to the reader what the relationship between them is.

Can this be made mode consistent? The surrounding text mostly uses just "sequence", sometimes as "source sequence".